guava/src/com/google/common/reflect/ClassPath.java

```diff
@@ -478,10 +484,18 @@ protected void scanJarFile(ClassLoader classloader, JarFile file) {
 
     @Override
     protected void scanDirectory(ClassLoader classloader, File directory) throws IOException {
-      scanDirectory(directory, classloader, "");
+      scanDirectory(directory, classloader, "", new HashSet<>());
     }
 
-    private void scanDirectory(File directory, ClassLoader classloader, String packagePrefix)
+    /**
+     * Recursively scan the given directory, adding resources for each file encountered.  Symlinks which have already
+     * been traversed in the current tree path will be skipped to eliminate cycles; otherwise symlinks are traversed.
+     * @param directory the root of the directory to scan
+     * @param classloader the classloader that includes resources found in {@code directory}
+     * @param packagePrefix resource path prefix inside {@code classloader} for any files found under {@code directory}
+     * @param currentPath canonical files already visited in the current directory tree path, for cycle elimination
+     */
+    private void scanDirectory(File directory, ClassLoader classloader, String packagePrefix, Set<File> currentPath)
```

Comment:
Is this method decl. <= 80 characters long? If not then running google-java-format should fix it. :)